app/products/calls/connection.ts

```diff
@@ -125,6 +125,10 @@ export async function newClient(channelID: string, closeCb: () => void, setScree
             return;
         }
 
+        if (await InCallManager.checkRecordPermission() !== 'granted') {
+            await InCallManager.requestRecordPermission();
```

Comment:
Do you think we should show anything if the permissions are not granted, even after prompting? E.g. returning an error.

Reply:
Yep, you're right -- I'll refactor this and use the same kind of feedback we use in the other permissions checks.

Reply:
There are some permissions checks already for stuff like accessing files in the code base, perhaps that is something you need to look into.

Also one thing that could happen is that the user denied the permission and now `request` won't show it again, so we need to identify that the permission has not been granted and probably redirect the user, to the App settings in the OS so they can enable it manually

Reply:
@enahum Done -- if the user accepts the permission, we start the call. If they deny, we don't start the call. If they try to join another call, we tell them we need permission and direct them to the settings page. If they accept the setting, they have to join the call again (and they can join). Tested on iPhone and Pixel.